src/components/dialogs/regulating-terminal-edition.js

```diff
@@ -99,26 +100,37 @@ const RegulatingTerminalEdition = ({
             );
         }
     }, [
-        regulatingTerminalValue.voltageLevel,
+        regulatingTerminalValue?.voltageLevel?.id,
         setEquipmentsOptions,
         voltageLevelEquipmentsCallback,
         voltageLevelsEquipments,
     ]);
 
     useEffect(() => {
+        let selectedExistingEquipment;
+        if (regulatingTerminalValue?.equipmentSection) {
+            selectedExistingEquipment = equipmentsOptions.find(
+                (value) =>
+                    value.id === regulatingTerminalValue.equipmentSection.id
+            );
+        }
         setCurrentEquipment(
             regulatingTerminalValue?.equipmentSection &&
-                equipmentsOptions.length
-                ? equipmentsOptions.find(
-                      (value) =>
-                          value.id ===
-                          regulatingTerminalValue.equipmentSection.id
-                  )
+                selectedExistingEquipment
+                ? selectedExistingEquipment
                 : regulatingTerminalValue?.equipmentSection === null
```

Comment:
Here we need to check explicitly for null value because it is the default/reset value
I've tried to do minimal changes to this component because it's really easily breakable. 
I reckon it'll need a good refactoring because in its current state it is quite complex to tweak